ReSharper.FSharp/src/FSharp.Psi/src/CodeFormatter/FSharpCodeFormatterInfoProvider.cs

```diff
@@ -12,8 +15,78 @@ public class FSharpFormatterInfoProvider :
   {
     public FSharpFormatterInfoProvider(ISettingsSchema settingsSchema) : base(settingsSchema)
     {
+      var bindingAndModuleDeclIndentingRulesParameters = new[]
+      {
+        ("NestedModuleDeclaration", ElementType.NESTED_MODULE_DECLARATION, NestedModuleDeclaration.MODULE_MEMBER),
+        ("TopBinding", ElementType.TOP_BINDING, TopBinding.CHAMELEON_EXPR),
+        ("LocalBinding", ElementType.LOCAL_BINDING, LocalBinding.EXPR),
+      };
+
+      var synExprIndentingRulesParameters = new[]
+      {
+        ("ForExpr", ElementType.FOR_EXPR, ForExpr.DO_EXPR),
+        ("ForEachExpr", ElementType.FOR_EACH_EXPR, ForEachExpr.DO_EXPR),
+        ("WhileExpr", ElementType.WHILE_EXPR, WhileExpr.DO_EXPR),
+        ("DoExpr", ElementType.DO_EXPR, DoExpr.EXPR),
+        ("AssertExpr", ElementType.ASSERT_EXPR, AssertExpr.EXPR),
+        ("LazyExpr", ElementType.LAZY_EXPR, LazyExpr.EXPR),
+        ("FixedExpr", ElementType.FIXED_EXPR, FixedExpr.EXPR),
+        ("InferredUpcastExpr", ElementType.INFERRED_UPCAST_EXPR, InferredUpcastExpr.EXPR),
+        ("InferredDowncastExpr", ElementType.INFERRED_DOWNCAST_EXPR, InferredDowncastExpr.EXPR),
+        ("ComputationExpr", ElementType.COMPUTATION_EXPR, ComputationExpr.EXPR),
+        ("LambdaExpr", ElementType.LAMBDA_EXPR, LambdaExpr.EXPR),
```

Comment:
Does it support deindenting like in the following code?

```fsharp
let foo x = x |> fun _ ->
    ()
```

Reply:
Yes.